src/shogun/base/SGObject.cpp

```diff
@@ -986,6 +988,27 @@ bool CSGObject::equals(const CSGObject* other) const
 	return true;
 }
 
+template <typename T>
+void CSGObject::for_each_param_of_type(
+    std::function<void(const std::string&, T*)> operation)
+{
+	auto visitor = std::make_unique<FilterVisitor<T>>(operation);
+	const auto& param_map = self->map;
+
+	std::for_each(param_map.begin(), param_map.end(), [&](auto& pair) {
+		Any any_param = pair.second.get_value();
```

Comment:
Any& ?

Reply:
Any& doesn't work
get_value returns Any

Reply:
shouldn't it return a reference as well?

Reply:
I think that's reasonable, should I change its implementation?

Reply:
i think so, but pls another pr so it is visible and we can discuss it with the others